File: src/main/java/com/novy/teamsmanagement/controller/TeamController.java
package com.novy.teamsmanagement.controller;

import com.novy.teamsmanagement.model.Team;
import com.novy.teamsmanagement.service.TeamService;
import org.springframework.beans.factory.annotation.Autowired;
import org.springframework.http.HttpStatus;
import org.springframework.http.MediaType;
import org.springframework.stereotype.Controller;
import org.springframework.web.bind.annotation.*;

import java.util.Collection;

/**
 * Created by novy on 18.10.14.
 */

@Controller
@RequestMapping("/teams")
public class TeamController {

    @Autowired
    private TeamService teamService;

    @RequestMapping(method = RequestMethod.GET, produces = MediaType.APPLICATION_JSON_VALUE)
    public @ResponseBody
    Collection<Team> viewAllTeams() {
        return teamService.findAll();
    }

    @RequestMapping(
            method = RequestMethod.POST,
            consumes = MediaType.APPLICATION_JSON_VALUE,
            produces = MediaType.APPLICATION_JSON_VALUE
    )
    @ResponseStatus(HttpStatus.CREATED)
    public void addNewTeam(@RequestBody Team team) {
        teamService.addTeam(team);
        // todo: should return Location: loc in response
    }

    @RequestMapping(
            value = "/{id}",
            method = RequestMethod.PUT,
            consumes = MediaType.APPLICATION_JSON_VALUE
    )
    @ResponseStatus(HttpStatus.NO_CONTENT)
    public void updateTeam(@PathVariable("id") Integer id, @RequestBody Team team) {
        team.setId(id);
        teamService.updateTeam(team);
    }

    @RequestMapping(value = "/{id}", method = RequestMethod.DELETE)
    @ResponseStatus(HttpStatus.NO_CONTENT)
    public void deleteTeam(@PathVariable("id") Integer id) {
        teamService.deleteTeam(id);
    }

    @RequestMapping("/layout")
    public String getTeamsPartialPage() {
        return "teams/layout";
    }
}

File: src/main/webapp/resources/js/controllers/TeamController.js
/**
 * Created by novy on 18.10.14.
 */

var teamController = function($scope, $http) {

    var viewAllTeams = function() {
        $http.get('teams').success(function(response) {
            $scope.teams = response;
        })
    };

    var clearFormTeam = function() {
        $scope.formTeam.name = '';
        $scope.formTeam.division = '';
        $scope.formTeam.rating = '';
    };

    var setEditionMode = function(editionMode) {
        $scope.editionMode = editionMode;
    };

    $scope.addTeam = function(team) {
        $http.post('teams', team).success(viewAllTeams)
    };

    $scope.updateTeam = function(team) {
        $http.put('teams/' + team.id, team).success(function(response) {
            viewAllTeams();
            clearFormTeam();
            setEditionMode(false);
        })
    };

    $scope.deleteTeam = function(team) {
        $http.delete('teams/' + team.id).success(viewAllTeams)
    };

    $scope.selectForEdition = function(team) {
        $scope.formTeam = {};
        $scope.formTeam.id = team.id;
        $scope.formTeam.name = team.name;
        $scope.formTeam.division = team.division;
        $scope.formTeam.rating = team.rating;
        setEditionMode(true);
    }

    viewAllTeams();
    setEditionMode(false)

};